client-core/src/synchronizer/manual_synchronizer.rs

```diff
@@ -195,7 +196,9 @@ where
                 }
             }
 
-            self.global_state_service.save_trust_state(&trust_state)?;
+            global_state.trusted_state = trusted_state;
+            self.global_state_service
```

Comment:
Because `trusted_state` is updated once for the whole batch, if we save other parts of the global state per block, it would not be consistent.
I think the chance of crash during the processing of a block batch which already downloaded from the network is quite low.

Reply:
Ok.